pkg/didcomm/protocol/middleware/presentproof/middlewares.go

```diff
@@ -103,11 +114,73 @@ type presentationExchangePayload struct {
 	PresentationDefinition *presexch.PresentationDefinition `json:"presentation_definition"`
 }
 
+// OptPD represents option function for the PresentationDefinition middleware.
+type OptPD func(o *pdOptions)
+
+// WithAddProofFn allows providing function that will sign the Presentation.
+func WithAddProofFn(sign func(presentation *verifiable.Presentation) error) OptPD {
+	return func(o *pdOptions) {
+		o.addProof = sign
+	}
+}
+
+type pdOptions struct {
+	addProof func(presentation *verifiable.Presentation) error
+}
+
+func defaultPdOptions() *pdOptions {
+	return &pdOptions{
+		addProof: func(presentation *verifiable.Presentation) error {
+			return nil
+		},
+	}
+}
+
+// AddBBSProofFn add BBS+ proof to the Presentation.
+func AddBBSProofFn(p Provider) func(presentation *verifiable.Presentation) error {
+	const bls12381g2pub = 0xeb
+
+	km, cr := p.KMS(), p.Crypto()
+
+	return func(presentation *verifiable.Presentation) error {
+		bbsLoader, err := bbsJSONLDDocumentLoader()
+		if err != nil {
+			return err
+		}
+
+		kid, pubKey, err := km.CreateAndExportPubKeyBytes(kms.BLS12381G2Type)
+		if err != nil {
+			return err
+		}
+
+		methodID := fingerprint.KeyFingerprint(bls12381g2pub, pubKey)
```

Comment:
@Baha-sk is going to add Create DID Key feature for bls12381 keys, please add a follow up task for him to replace this code with create bbs DID key api.

Reply:
Done https://github.com/hyperledger/aries-framework-go/pull/2601